### PR TITLE
feat(cds-1003): button group component designed

### PR DESCRIPTION
**Pull request by volkangurenekli**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,6 +46,7 @@
     "eslint-plugin-prettier": "^4.0.0",
     "husky": "^7.0.4",
     "lint-staged": "^12.3.7",
+    "ng-mocks": "^13.4.2",
```
Comment (ahmetatar) [inline package.json:49]: Before deciding to use a new package, we need to discuss together ;)

File: src/app/shared/components/button-group/button-group.component.scss [hunk 2]
```diff
@@ -0,0 +1,21 @@
+.btn-group desx-button .btn {
+  border-top-left-radius: 0;
```
Comment (ahmetatar) [inline src/app/shared/components/button-group/button-group.component.scss:2]: Why do you need to do something like this for buttons? Doesn't the button group already do this? Also all css rules should be gathered under desx-button-group.

Comment (ahmetatar) [inline src/app/shared/components/button-group/button-group.component.scss:2]: Why do you need to do something like this for buttons? Doesn't the button group already do this? Also all css rules should be gathered under desx-button-group.

File: src/app/shared/components/button-group/button-group.component.spec.ts [hunk 3]
```diff
@@ -1,24 +1,80 @@
...
+    component.btnGroupStyle = ButtonGroupStyles.Vertical;
+    fixture.detectChanges();
+
+    const buttonGroup = buttonGroupComponentFixture.getVerticalButtonGroup();
+
+    expect(buttonGroup).toBeTruthy();
+  });
+
+  it('creates button group only with desx-button', () => {
+    const rawHtml = MockRender<ButtonComponent>(
```
Comment (ahmetatar) [inline src/app/shared/components/button-group/button-group.component.spec.ts:67]: Why did you need to do such a thing?

Comment (ahmetatar) [inline src/app/shared/components/button-group/button-group.component.spec.ts:67]: Why did you need to do such a thing?

**Review by ahmetatar — CHANGES_REQUESTED** [review]
Looks good but you need to change somethings 👍

File: package.json (hunk 1 above)
Comment (volkangurenekli) [reply package.json]: ng-mocks library removed

File: src/app/shared/components/button-group/button-group.component.scss (hunk 2 above)
Comment (volkangurenekli) [reply src/app/shared/components/button-group/button-group.component.scss]: fixed

File: src/app/shared/components/button-group/button-group.component.spec.ts (hunk 3 above)
Comment (volkangurenekli) [reply src/app/shared/components/button-group/button-group.component.spec.ts]: MockRender function removed

Comment (volkangurenekli) [reply src/app/shared/components/button-group/button-group.component.spec.ts]: MockRender function removed

File: src/app/shared/components/button-group/button-group.component.spec.ts [hunk 4]
```diff
@@ -1,24 +1,112 @@
...
+
+    const buttonGroup = buttonGroupComponentFixture.getVerticalButtonGroup();
+
+    expect(buttonGroup).toBeTruthy();
+  });
+
+  it('creates button group only with desx-button', () => {
+    fixture.detectChanges();
+
+    const buttonGroup = buttonGroupComponentFixture.getButtonGroup()?.innerHTML;
```
Comment (ahmetatar) [inline src/app/shared/components/button-group/button-group.component.spec.ts:79]: Here you can have an undefined value. So you should have an expectation to check this

`expect(buttonGroup).toBeTruthy();`

**Review by ahmetatar — CHANGES_REQUESTED** [review]
One last change :)

File: src/app/shared/components/button-group/button-group.component.spec.ts (hunk 4 above)
Comment (volkangurenekli) [reply src/app/shared/components/button-group/button-group.component.spec.ts]: fixed

**Pull request merged**